File: ios/ObjectiveC/Beacon/2-Notification/Complete/NotificationComplete/AsyncNotificationTask.m
//
//  AsyncNotificationTask.m
//  NotificationComplete
//
//  Created by Ludovic Vimont on 27/10/2017.
//  Copyright © 2017 sarra srairi. All rights reserved.
//

#import "AsyncNotificationTask.h"
 
@import ConnectPlaceActions;

@implementation AsyncNotificationTask


- (void)launchNotificationTaskWithPlaceNotification:(AdtagPlaceNotification * _Nonnull)placeNotification displayPlaceNotificationDelegate:(id<DisplayPlaceNotificationDelegate> _Nonnull)displayPlaceNotificationDelegate {
    [displayPlaceNotificationDelegate displayPlaceNotification:[[PlaceNotificationImage alloc] initWithPlaceNotification:placeNotification]];
}

@end

File: ios/SWIFT/Beacon/2-Notification/Complete/Complete/AsyncNotificationTask.swift
//
//  AsyncNotificationTask.swift
//  Complete
//
//  Created by Ludovic Vimont on 27/10/2017.
//  Copyright © 2017 R&D connecthings. All rights reserved.
//

import Foundation
import PromiseKit
import ConnectPlaceActions
import AdtagConnection
import AdtagLocationBeacon

public class AsyncNotificationTask: NSObject, AdtagNotificationTaskDelegate {
    public func launchNotificationTask(placeNotification: AdtagPlaceNotification, displayPlaceNotificationDelegate: DisplayPlaceNotificationDelegate) {
        displayPlaceNotificationDelegate.displayPlaceNotification(PlaceNotificationImage(placeNotification: placeNotification))
    }
}
